imputed MKT extension

osipovarev opened the pull request:
1) added recalculation of allele frequency based on ancestral allele in polarized VCF
2) fixed a bug causing pN pS undercounting
3) added imputed MKT extension

gwct commented on degenotate_lib/degen.py:
```diff
@@ -248,6 +293,7 @@ def processCodons(globs):
                 if frame != 0:
                     for out_of_frame_pos in range(extra_leading_nt):
                         outline = OUT.compileBedLine(globs, transcript, transcript_region, cds_coord, globs['cds-seqs'][transcript][cds_coord], "", "", ".", ".", "");
+                        print('OUTLINE: {}'.format(outline))
```
Should this print statement still be here or is it just for debugging?

gwct commented on degenotate_lib/degen.py:
```diff
@@ -346,20 +396,24 @@ def processCodons(globs):
-
-                        for poly_codon in mk_alleles['poly']:
-                        # For in group variants, we treat each SNP as independent
+                        print(mk_alleles)
```
Another print statement

gwct commented on degenotate_lib/degen.py:
```diff
@@ -426,18 +476,61 @@ def processCodons(globs):
+                p0_high = len(ps_af_high)
+                p_low = len(pn_af_low)
+                p0_low = len(ps_af_low)
+                print('p_high = {}, p_low = {}, p0_high = {}, p0_low = {}'.format(p_high, p_low, p0_high, p0_low))
```
Another print statement

gwct commented on degenotate_lib/degen.py:
```diff
@@ -426,18 +476,61 @@ def processCodons(globs):
+
+                d = transcript_output['mk']['dn']
+                d0 = transcript_output['mk']['ds']
+                print('for MKT: dn = {}; ds = {}'.format(d, d0))
```
Another print statement

gwct commented on degenotate_lib/degen.py:
```diff
@@ -426,18 +476,61 @@ def processCodons(globs):
+                transcript_output['mk']['pval'] = ext_pval
+                transcript_output['mk']['imp_pval'] = imp_pval
+
+                print(transcript_output['mk'])
```
Another print statement

gwct commented on degenotate_lib/vcf.py:
```diff
@@ -42,6 +42,12 @@ def read(globs):
 
+    if 'AA' in globs['vcf'].header.info:
+        globs['vcf-polarized'] = True
+        CORE.printWrite(globs['logfilename'], 1, "# WARNING: " + "provided VCF appears to be polarized (has ancestral allele field in the header) => will try to recalculate derived allele frequency and run imputed MKT framework.");
```
Maybe I'm getting this wrong, but if the warning is here with `1` as the verbosity option to `printWrite()`, it will be printed to the screen and mess up the status report later on. (I think the options for `printWrite()` are kind of messed up now that I'm looking at it, but that's another issue.) If we want to not mess up the status report in the shell, we should just have a flag here that is returned to [degenotate.py](https://github.com/harvardinformatics/degenotate/blob/4cf3758b7d7434682fbb53a41588390a83e853f7/degenotate.py#L78) and [printed out after the status update on the next line](https://github.com/harvardinformatics/degenotate/blob/4cf3758b7d7434682fbb53a41588390a83e853f7/degenotate.py#L79).

gwct reviewed (changes requested):
I noted several places that had print statements left in that I think are for debugging and should be commented out/removed before the next release.

I'm also wondering if some of the original things are still calculated/reported. For instance, are pN and pS still written to the transcript output? And are the original MK statistics still reported or just the imputed ones?

osipovarev commented:
1) removed all the debugging prints
2) corrected output to have both standard and imputed MK tables (if impMKT is run)
3) edited the warning about polarized VCF so it goes to log only.

gwct commented on degenotate_lib/opt_parse.py:
```diff
@@ -434,6 +446,13 @@ def startProg(globs):
                         "These samples will be used as outgroups in the VCF file and all others as ingroups.");
             # Report VCF outgroup samples
 
+            if globs['vcf-polarized']:
```
Maybe I'm missing something, but it looks like we removed the `-p` option above to specify whether the vcf is polarized. Instead it looks as though this is checked later (in `vcf.py`). If that's correct, then this will never be True here so this whole if statement could be removed. But if we want to keep it just in case that's fine.

osipovarev replied:
yes, you're right, this one shouldn't be ever executed, I just forgot to remove it after I removed -p flag.
Cleaned it up now.
